File: model/src/main/java/no/nav/pensjon/vtp/testmodell/ansatt/AnsatteIndeks.java
package no.nav.pensjon.vtp.testmodell.ansatt;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.stream.Stream;

public class AnsatteIndeks {
    private List<NAVAnsatt> ansatte = new ArrayList<>();

    public Stream<NAVAnsatt> hentAlleAnsatte() {
        return ansatte.stream();
    }

    public void leggTil(List<NAVAnsatt> ansatte) throws RuntimeException {
        this.ansatte.addAll(ansatte);
    }

    public Optional<NAVAnsatt> hentNAVAnsatt(String ident) {
        return ansatte.stream().filter(ansatt -> ansatt.cn.equals(ident)).findFirst();
    }
}

File: model/src/main/java/no/nav/pensjon/vtp/testmodell/ansatt/AnsatteIndeks.kt
package no.nav.pensjon.vtp.testmodell.ansatt

import java.util.*

class AnsatteIndeks {
    private val ansatte: MutableList<NAVAnsatt> = ArrayList()

    fun findAll(): List<NAVAnsatt> {
        return ansatte
    }

    fun saveAll(ansatte: List<NAVAnsatt>) {
        this.ansatte.addAll(ansatte)
    }

    fun findByCn(ident: String): NAVAnsatt? {
        return ansatte.firstOrNull { it.cn == ident }
    }

    fun findBySnIgnoreCase(sn: String): NAVAnsatt? {
        return ansatte.firstOrNull { sn.equals(it.sn, ignoreCase = true) }
    }

    fun findByEnhetsId(enhetsId: Long): List<NAVAnsatt> {
        return ansatte.filter { it.enheter.contains(enhetsId) }
    }
}
